test/model-serialization.test.js

```diff
@@ -113,16 +113,20 @@ test('model serialization', () => {
   for (let i = 0; i < 200; i++) {
     model1.step(TIMESTEP)
   }
-  const model2 = Model.deserialize(model1.serialize())
+  const model1Data = model1.serialize()
+  const model2 = Model.deserialize(model1Data)
   compareModels(model2, model1)
 
   // Make sure that all the properties have been serialized and deserialized correctly.
   // Comparision after next 200 steps should ensure that models are identical.
   for (let i = 0; i < 200; i++) {
-    // Mock Math.random as model progress depends on it (islands generation).
-    Math.random = () => { return i / 200 }
     model1.step(TIMESTEP)
-    model2.step(TIMESTEP)
   }
-  compareModels(model2, model1)
+  // We can't use just one loop as random seed is set when model is created or deserialized. Creating model at this
+  // point ensures that both use the same seedrandom state.
+  const model3 = Model.deserialize(model1Data)
+  for (let i = 0; i < 200; i++) {
+    model3.step(TIMESTEP)
+  }
+  compareModels(model3, model1)
```

Comment:
Should this be pulled out into a separate unit test, to check the randomness specifically?

Reply:
This test both checks randomness, but also it checks whether models have been serialized and deserialized correctly. If some property is missing, it's very likely the models will diverge after some time. So, the main purpose of this test is to check serialization. I thought about dividing it, but it's difficult to do that without making tests slower (and they're already pretty slow).